ui/app/components/clients/date-range.hbs

```diff
@@ -0,0 +1,92 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+<div ...attributes>
+  <Hds::Text::Display @tag="p" class="has-bottom-margin-xs">
+    Date range
+  </Hds::Text::Display>
+
+  <div class="is-flex-align-baseline">
+    {{#if (or @startTime @endTime)}}
```

Comment:
I think this conditional should be `and`. The client count dashboard is currently still accessible on CE, but we don't have license data so the workflow now is the counts route prompts the user to select a "Client counting start date" first. The empty state used to be clearer, but this is how it looks now. (I mocked this by just changing the version service getters to return `true` for community and `false` for enterprise). _Technically_ the activity config does return a `billing_start_timestamp` for community but it's the `go` default for "no date" which is `"0001-01-01T00:00:00Z"`

<img width="1093" alt="Screenshot 2024-07-16 at 3 16 37 PM" src="https://github.com/user-attachments/assets/323e0a58-4197-4fd5-9853-7fa57aecec8e">

I'm not sure what the test coverage is around this. There are so many client count states 😢 

I wonder if it'd be useful in the `{{else}}` conditional instead of assuming "using default date range" we show some sort of zero state and have users explicitly set the start/end? (I checked out your branch to test, and this is what I'm seeing when no start is available now)
<img width="1060" alt="Screenshot 2024-07-16 at 3 22 48 PM" src="https://github.com/user-attachments/assets/d1fd9a46-9572-4e69-887f-afcf554b531e">

Reply:
Initially I had in mind that we were going to remove the initial no data state from CE with the next phase of work, but I'm going to go ahead and make the date range component do something similar where it prompts to set the range unless both start and end are passed 👍